x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeListener.java

```diff
@@ -29,30 +32,71 @@
  * 2. Collects driver profiles from sub tasks.
  * 3. Collects response headers from sub tasks, specifically warnings emitted during compute
  * 4. Collects failures and returns the most appropriate exception to the caller.
+ * 5. Updates {@link EsqlExecutionInfo} for display in the response for cross-cluster searches
  */
 final class ComputeListener implements Releasable {
     private static final Logger LOGGER = LogManager.getLogger(ComputeService.class);
 
     private final RefCountingListener refs;
+    // MP TODO: what is this failureCollector for? should I use it for CCS metadata?
     private final FailureCollector failureCollector = new FailureCollector();
     private final AtomicBoolean cancelled = new AtomicBoolean();
     private final CancellableTask task;
     private final TransportService transportService;
     private final List<DriverProfile> collectedProfiles;
     private final ResponseHeadersCollector responseHeaders;
 
-    ComputeListener(TransportService transportService, CancellableTask task, ActionListener<ComputeResponse> delegate) {
+    private final EsqlExecutionInfo esqlExecutionInfo;
+
+    // for use by top level ComputeListener
+    ComputeListener(
```

Comment:
I think we should make a `private` ctor and make these named static methods.

Reply:
Thanks Nik. Change made in [this commit](https://github.com/elastic/elasticsearch/pull/112472/commits/cd37244aab6fdb85d911da32db644c5fff70f296).